{"version":3,"sources":["../../app-src/util/ApplicationException.js"],"names":["ApplicationException","Error","constructor","msg","name","exception","isApplicationException","err","Object","getPrototypeOf","getExceptionMessage","message","console","log"],"mappings":";;;;;;AAAO,kBAAMA,oBAAN,SAAmCC,KAAnC,CAAyC;AAC5CC,4BAAYC,MAAM,EAAlB,EAAsB;AAClB,0BAAMA,GAAN;AACA,yBAAKC,IAAL,GAAY,KAAKF,WAAL,CAAiBE,IAA7B;AACH;AAJ2C;;;;AAOhD,kBAAMC,YAAYL,oBAAlB;;AAEO,qBAASM,sBAAT,CAAgCC,GAAhC,EAAqC;AACxC,uBAAOA,eAAeF,SAAf,IACHG,OAAOC,cAAP,CAAsBF,GAAtB,aAAsCF,SAD1C;AAEH;;;;AAEM,qBAASK,mBAAT,CAA6BH,GAA7B,EAAkC;AACrC,oBAAGD,uBAAuBC,GAAvB,CAAH,EAA+B;AAC3B,2BAAOA,IAAII,OAAX;AACH,iBAFD,MAEO;AACHC,4BAAQC,GAAR,CAAYN,GAAZ;AACA,2BAAO,wCAAP;AACH;AACJ","file":"ApplicationException.js","sourcesContent":["export class ApplicationException extends Error {\r\n    constructor(msg = '') {\r\n        super(msg);\r\n        this.name = this.constructor.name;\r\n    }\r\n}\r\n\r\nconst exception = ApplicationException;\r\n\r\nexport function isApplicationException(err) {\r\n    return err instanceof exception ||\r\n        Object.getPrototypeOf(err) instanceof exception\r\n}\r\n\r\nexport function getExceptionMessage(err) {\r\n    if(isApplicationException(err)){\r\n        return err.message;\r\n    } else {\r\n        console.log(err);\r\n        return 'Não Foi Possível Realizar a Opereação.';\r\n    }\r\n}"]}